features/5-basic-sso/api.md

```diff
@@ -0,0 +1,254 @@
+# API Design Overview
+
+Have official plugins for common SSO support; And also make it easy to enable and config them from Skygear Portal.
+
+Portal needs to have an interface for on/off and configuration of ID / Secret.
+
+## Scenario
+
+* Sign in with web popup
+* Sign in with web redirect
+* Sign in with limited capability devices (TV/Command line)
+* Sign in with Mobile
+* Login by `AuthProvider A`, but the email already got another account from `AuthProvider B`, show an error and tell users to login with another `AuthProvide B`
+* Login by `AuthProvider A`, but the email already got another account from `AuthProvider B`, tell users to login and link with new AuthProvider
+* Login by `AuthProvider A`, but the email already got another account from `AuthProvider B`, assume it is two different accounts (will break the assumption of Skygear, which each users got unique email address)
+* At User Setting page, add another `AuthProvider`
```

Comment:
As discussed in Product Meeting, will need a use case/samples code of using oauth_token in Cloud Functions.